examples/basic_authentication/lib/user_repository.dart

```diff
@@ -0,0 +1,128 @@
+import 'dart:convert';
+
+import 'package:crypto/crypto.dart';
+import 'package:equatable/equatable.dart';
+import 'package:meta/meta.dart';
+
+final List<User> _db = [];
```

Comment:
Related to https://github.com/VeryGoodOpenSource/dart_frog/pull/715#discussion_r1239565037.

Reply:
Does it matter in this case? This is just a fictional database 😉

Reply:
That's true as you point out the performance in this fictional database is not of a massive worry, since it is an example.   I should have been more clear with the comment. Since my intention pointing out that fact wasn't precisely improving the performance of the example, but leading by example.  Let me rephrase:

I believe the objective of the example is to show users how the tool works without confusion and also take the opportunity to lead and inculcate good practices and behaviours.

Firstly, I think that changing the List to a Map will avoid readers feeling confused when reading the example, here's why:

Throughout the code you'll find that an `id` variable is being assigned when creating users, see [here](https://github.com/VeryGoodOpenSource/dart_frog/blob/f674f3e314b06f416373aa7e5475c3704c0b8f96/examples/basic_authentication/routes/users/index.dart#L22). Developers usually expect "id" to refer to "identifier" and usually identifiers are unique. This leads to the developer thinking that users are unique within the example. However, unintuitively in this example, you can have multiple instances of the same user (with the same name and id) with multiple different passwords! Personally, I think this will confuse some of our readers and using a `Map`-like data structure will solve this strange behaviour.

Secondly, I think that we should try to lead by example and show off good practices in our examples to. In this case, the List is not an appropriate data structure since we are only using it to search within it. I think is a nice bonus to inculcate that a Map is a clearly a better choice for the given scope of the example.